packages/iov-bcp-types/src/atomicswap.ts

```diff
@@ -0,0 +1,94 @@
+import { Stream } from "xstream";
+
+import { BcpCoin, BcpConnection, BcpQueryEnvelope } from "./bcp";
+import { Address } from "./signables";
+import { SwapIdBytes } from "./transactions";
+
+export enum SwapState {
+  OPEN = "open",
+  CLAIMED = "claimed",
+  EXPIRED = "expired",
+}
+
+export interface SwapData {
+  readonly id: SwapIdBytes; // this is used as an unique identitier to locate the swap
+  readonly sender: Address;
+  readonly recipient: Address;
+  readonly hashlock: Uint8Array; // this is the hash, whose preimage releases the swap
+  readonly amount: ReadonlyArray<BcpCoin>;
+  readonly timeout: number; // blockheight where the swap expires (TODO: alternatively support Date?)
```

Comment:
I am not sure how doing it by date would work. We can simulate it on the frontend already for user inputs, as we have a block time estimator.

Reply:
The date would have to be implemented on the blockchain. Since it enforces the expiration.

Right now, we just use blockheight in weave. A note to reconsider if we encounter chains (or update weave), so they use wall time to control expiration.